libraries/AP_HAL_ChibiOS/hwdef/PixC4-Jetson/hwdef.dat

```diff
@@ -0,0 +1,326 @@
+PB6 USART1_TX USART1 NODMA
+
+# USART3 is telem2
+PD9 USART3_RX USART3
+PD8 USART3_TX USART3
+PD11 USART3_CTS USART3
+PD12 USART3_RTS USART3
+
+# UART4 GPS2
+PD0 UART4_RX UART4 NODMA
+PD1 UART4_TX UART4 NODMA
+
+# USART6 is telem3
+PG9 USART6_RX USART6 NODMA
+# we leave PG14 as an input to prevent it acting as a pullup
+# on the IOMCU SBUS input
+# PG14 USART6_TX USART6 NODMA
+PG15 USART6_CTS USART6
+PG8 USART6_RTS USART6
+
+# UART7 is debug
+PF6 UART7_RX UART7 NODMA
+PE8 UART7_TX UART7 NODMA
+
+# UART8 is for IOMCU
+PE0 UART8_RX UART8
+PE1 UART8_TX UART8
+
+# UART for IOMCU
+IOMCU_UART UART8
+
+# enable RTSCTS support
+define AP_FEATURE_RTSCTS 1
+
+# enable SBUS_OUT on IOMCU (if you have an IOMCU)
+define AP_FEATURE_SBUS_OUT 1
+
+# PWM AUX channels
+PE14 TIM1_CH4 TIM1 PWM(1) GPIO(50)
+PA10 TIM1_CH3 TIM1 PWM(2) GPIO(51)
+PE11 TIM1_CH2 TIM1 PWM(3) GPIO(52)
+PE9  TIM1_CH1 TIM1 PWM(4) GPIO(53)
+PD13 TIM4_CH2 TIM4 PWM(5) GPIO(54)
+PD14 TIM4_CH3 TIM4 PWM(6) GPIO(55)
+# we need to disable DMA on the last 2 FMU channels
+# as timer 12 doesn't have a TIMn_UP DMA option
+PH6  TIM12_CH1 TIM12 PWM(7) GPIO(56) NODMA
+PH9  TIM12_CH2 TIM12 PWM(8) GPIO(57) NODMA
+
+define BOARD_PWM_COUNT_DEFAULT 8
+
+# PWM output for buzzer
+PE5 TIM15_CH1 TIM15 GPIO(77) ALARM
+
+# analog in
+PA0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PA1 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+PA2 BATT2_VOLTAGE_SENS ADC1 SCALE(1)
+PA3 BATT2_CURRENT_SENS ADC1 SCALE(1)
+
+PC4 SPARE1_ADC1 ADC1 SCALE(1)
+PA4 SPARE2_ADC1 ADC1 SCALE(1)
+
+PB0 RSSI_IN ADC1 SCALE(1)
+
+PC0 VDD_5V_SENS ADC1 SCALE(2)
+PC1 SCALED_V3V3 ADC1 SCALE(2)
+
+# setup scaling defaults
+define HAL_BATT_VOLT_SCALE 18.0
+define HAL_BATT_CURR_SCALE 24.0
+define HAL_BATT_VOLT_PIN 0
+define HAL_BATT_CURR_PIN 1
+define HAL_BATT2_VOLT_PIN 2
+define HAL_BATT2_CURR_PIN 3
+
+# CAN bus
+PI9  CAN1_RX CAN1
+PH13 CAN1_TX CAN1
+
+PB12 CAN2_RX CAN2
+PB13 CAN2_TX CAN2
+
+PH2 GPIO_CAN1_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(70)
+PH3 GPIO_CAN2_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(71)
+PH4 GPIO_CAN3_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(72)
+
+# GPIOs
+PA7 HEATER_EN OUTPUT LOW GPIO(80)
+define HAL_HEATER_GPIO_PIN 80
+
+PG1 VDD_BRICK_VALID INPUT PULLUP
+PG2 VDD_BRICK2_VALID INPUT PULLUP
+PG3 nVBUS INPUT PULLDOWN
+PA9 VBUS INPUT
+PF13 VDD_5V_HIPOWER_OC INPUT PULLUP
+PE15 VDD_5V_PERIPH_OC INPUT PULLUP
+
+# capture pins
+PA5 FMU_CAP1 INPUT  GPIO(58)
+PB3 FMU_CAP2 INPUT  GPIO(59)
+PB11 FMU_CAP3 INPUT GPIO(60)
+PI0 FMU_SPARE_4 INPUT GPIO(61)
+
+# SPI devices
+# note that SPI6 is disabled by default, try to enable it once we get one sensor up
+SPIDEV ms5611	       SPI4 DEVID5  BARO_CS    MODE3  5*MHZ  5*MHZ
+SPIDEV icm20602        SPI1 DEVID1  ICM20602_CS  MODE3  2*MHZ  8*MHZ
+SPIDEV icm20602_2      SPI1 DEVID2  ICM20602_CS2  MODE3  2*MHZ  8*MHZ
+SPIDEV icm42688        SPI1 DEVID3  ICM42688_CS  MODE3  2*MHZ  8*MHZ
+SPIDEV ramtron         SPI2 DEVID1  FRAM_CS      MODE3  8*MHZ  8*MHZ 
+
+# 3 IMUs
+IMU Invensense SPI:icm20602 ROTATION_YAW_90
```

Comment:
the 42688 is likely to perform best, so should probably be first IMU

Reply:
done